ci-operator/step-registry/cert-manager/install/cert-manager-install-commands.sh

```diff
@@ -4,97 +4,124 @@ set -e
 set -u
 set -o pipefail
 
+timestamp() {
+    date -u --rfc-3339=seconds
+}
+
 if [ -f "${SHARED_DIR}/proxy-conf.sh" ] ; then
     source "${SHARED_DIR}/proxy-conf.sh"
     echo "proxy: ${SHARED_DIR}/proxy-conf.sh"
 fi
 
-CATSRC=qe-app-registry
-if [[ ! "$(oc get catalogsource qe-app-registry -n openshift-marketplace -o yaml)" =~ "lastObservedState: READY" ]]; then
-    echo "The catalogsource qe-app-registry is either not existing or not ready. Will use redhat-operators to install cert-manager Operator."
-    CATSRC=redhat-operators
+# Check if the catalogsource is ready to use
+if [[ ! "$(oc get catalogsource $INSTALL_CATSRC -n openshift-marketplace -o yaml)" =~ "lastObservedState: READY" ]]; then
+    echo "Skipping as the catalogsource $INSTALL_CATSRC is either not existing or not ready."
+    exit 0
```

Comment:
Should it be exit 1?

Reply:
Suggest the logic to be:
If $JOB_NAME does not contain operator-upgrade, use my original lines, i.e. switch to redhat-operators when qe-app-registry has issue.
Otherwise, use $INSTALL_CATSRC

Reply:
As per current design, we provide the ENV configs (INSTALL_CATSRC, UPGRADE_CATSRC) for users to set the specific catalogsource. So auto-switching is not the expected behavior.

Reply:
You can:
If INSTALL_CATSRC env var is specified, use that, otherwise use my original lines